File: app/src/main/java/com/codepath/debuggingchallenges/activities/CurrentDayActivity.java
package com.codepath.debuggingchallenges.activities;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import com.codepath.debuggingchallenges.R;

import java.util.Calendar;

public class CurrentDayActivity extends AppCompatActivity {

    TextView tvDay;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_current_day);
        tvDay = (TextView) findViewById(R.id.tvDay);
        tvDay.setText(getDayOfMonth());
    }

    private int getDayOfMonth() {
        Calendar cal = Calendar.getInstance();
        return cal.get(Calendar.DAY_OF_MONTH);
    }
}

File: app/src/main/java/com/codepath/debuggingchallenges/activities/CurrentDayActivity.kt
package com.codepath.debuggingchallenges.activities

import androidx.appcompat.app.AppCompatActivity
import android.widget.TextView
import android.os.Bundle
import android.view.View
import com.codepath.debuggingchallenges.R
import java.util.*

class CurrentDayActivity : AppCompatActivity() {
    var tvDay: TextView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_current_day)
        tvDay = findViewById<View>(R.id.tvDay) as TextView
        tvDay?.text = dayOfMonth.toString()
    }

    private val dayOfMonth: Int
        get() {
            val cal = Calendar.getInstance()
            return cal[Calendar.DAY_OF_MONTH]
        }
}